src/main/java/kg/jarkyn/server/RequestingSocket.java

```diff
@@ -0,0 +1,33 @@
+package kg.jarkyn.server;
+
+import kg.jarkyn.server.Incoming.Requester;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class RequestingSocket implements Requester {
+    private Socket socket;
+
+    public RequestingSocket(Socket socket) {
+        this.socket = socket;
+    }
+
+    public Socket getSocket() {
+        return socket;
+    }
+
+    public InputStream getInputStream() {
```

Comment:
If you're using IntelliJ, use Command+O (or was it Option+O? - don't remember) to override methods. Not only will it make you faster, because you don't have to type everything, it will also automatically add the `@Override` annotations.